### DISCUSSION TITLE
Add Current Library to library list

### DISCUSSION THREAD

**Discussion by JonFP** [codefori/vscode-ibmi#751] [General]
Current library is correctly used in all compiles etc. but does not appear in the library list.

**Comment by worksofliam** [comment]
Can you see the library list from the spool file to check it's not in there? I will double check a bit later also.

**Comment by JonFP** [comment]
The curly is specified on the user profile.  If it was not in the active list the compiles would fail.  

Spool shows:

Current library: KEHECLASS
Library list: QGPL QTEMP
Commands:
		CRTBNDRPG PGM(KEHECLASS/CRTRPT15) SRCFILE(KEHECLASS/QRPGLESRC) OPTION(*EVENTF) DBGVIEW(*ALL) TGTRLS(*CURRENT)

**Comment by worksofliam** [comment]
I just checked, and the current library is on the library list when commands are executed.

```
Current library: MANZAN
Library list: QGPL QTEMP QDEVELOP QBLDSYS QBLDSYSR SOTEST
Commands:
		DSPLIBL

CPD4090:  Printer device PRT01 not found. Output queue changed to QPRINT in library QGPL.
  5770SS1 V7R3M0  160422                    Library List                                          7/12/22 12:01:48        Page    1
                           ASP
    Library     Type       Device      Text Description
    QSYS        SYS                    System Library
    QSYS2       SYS                    System Library for CPI's
    QHLPSYS     SYS
    QUSRSYS     SYS                    System Library for Users
    QSHELL      PRD
    MANZAN      CUR
    QGPL        USR                    General Purpose Library
    QTEMP       USR
    QDEVELOP    USR
    QBLDSYS     USR
    QBLDSYSR    USR
    SOTEST      USR
                           * * * * *  E N D  O F  L I S T I N G  * * * * *
```

**Comment by JonFP** [comment]
But it doesn't show yup ion the Liblist in the right hand pane in the GUI.  that's what I meant.

The LIBL shown is misleading as it doesn't include the CRLIB.


Jon Paris

***@***.***

**Reply by m-tyler** [reply]
I believe CURLIB is set when you run commands on the server and is obtained from the CURLIB settings in the extension.   Using Ctrl+, to open the setting view, click the link to edit in settings.json.  Find `currentLibrary`.  For me it is set to my developer library.

**Reply by JonFP** [reply]
It is indeed - as I said from the outset it is _used_ in the compiles - but it is not _listed_ in the list of libraries shown on the LHS of the workbench.

**Comment by chrjorgensen** [comment]
@worksofliam @JonFP I have created a PR #837 to provide a solution to your request. Please check it out...

**Reply by JonFP** [reply]
Looks good - great approach.
